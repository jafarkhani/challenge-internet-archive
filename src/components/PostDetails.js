
function PostDetails(props){

    const exceptionKeys = ["title", "description", "backup_location", "licenseurl"];

    let rows = Object.keys(props.metadata)
        .filter(key => exceptionKeys.indexOf(key) === -1 )
        .map(key => 
            <tr>
                <td className="post-metadata-key">{key}</td>
                <td className="post-metadata-value">{props.metadata[key]}</td>
            </tr> 
        );
    
    return (
        <table>
            {rows}
        </table>
    );
}

export default PostDetails;