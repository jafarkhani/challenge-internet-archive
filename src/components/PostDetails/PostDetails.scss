
  .post-metadata-value{
    color: gray;
  }
  