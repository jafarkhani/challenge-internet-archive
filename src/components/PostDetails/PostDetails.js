
import './PostDetails.scss';

function PostDetails(props){

    const exceptionKeys = ["title", "description", "backup_location", "licenseurl"];

    let rows = Object.keys(props.metadata)
        .filter(key => exceptionKeys.indexOf(key) === -1 )
        .map(key => 
            <tr key={key}>
                <td className="post-metadata-key">{key}</td>
                <td className="post-metadata-value">{props.metadata[key]}</td>
            </tr> 
        );
        
    return (
        <table>
            {rows}
        </table>
    );
}

export default PostDetails;