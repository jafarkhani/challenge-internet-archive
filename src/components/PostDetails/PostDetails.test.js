import React from "react";
import renderer from 'react-test-renderer';

import PostDetails from "./PostDetails";

describe("PostDetails", ()=> {

    it('snapshot metadata post details', () => {
        const tree = renderer.create(<PostDetails metadata={{key1: "data1"}} />).toJSON();
        expect(tree).toMatchSnapshot();         
    });

})


