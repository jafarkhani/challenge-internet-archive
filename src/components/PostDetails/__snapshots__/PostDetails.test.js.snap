// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostDetails render metadata post details 1`] = `
<table>
  <tr>
    <td
      className="post-metadata-key"
    >
      key1
    </td>
    <td
      className="post-metadata-value"
    >
      data1
    </td>
  </tr>
</table>
`;

exports[`PostDetails snapshop metadata post details 1`] = `
<table>
  <tr>
    <td
      className="post-metadata-key"
    >
      key1
    </td>
    <td
      className="post-metadata-value"
    >
      data1
    </td>
  </tr>
</table>
`;

exports[`PostDetails snapshot metadata post details 1`] = `
<table>
  <tr>
    <td
      className="post-metadata-key"
    >
      key1
    </td>
    <td
      className="post-metadata-value"
    >
      data1
    </td>
  </tr>
</table>
`;
